src/presence/PresenceHeartbeat.js

```diff
@@ -0,0 +1,66 @@
+// @flow strict-local
+import { PureComponent } from 'react';
+import { AppState } from 'react-native';
+import type { Auth, Dispatch } from '../types';
+
+import { connect } from '../react-redux';
+import { tryGetAuth } from '../account/accountsSelectors';
+import { reportPresence } from '../actions';
+import Heartbeat from './heartbeat';
+
+type Props = $ReadOnly<{|
+  dispatch: Dispatch,
+  auth: Auth | void,
+|}>;
+
+/**
+ * Component providing a recurrent `presence` signal.
+ */
+// Note that "PureComponent" is of questionable veracity: this component's
+// entire purpose is to emit network calls for their observable side effects
+// as a side effect of being rendered.
+//
+// However, it is at least true that there is never a need to call `render()`
+// (and therefore `componentDidUpdate()`) if the props haven't changed.
+class PresenceHeartbeat extends PureComponent<Props> {
```

Comment:
On the previous PR we discussed a previous version of this comment: https://github.com/zulip/zulip-mobile/pull/3704#discussion_r352962392 Thanks for the updates!

I think it's pretty reasonable to say that the term "pure" here is confusing and of questionable veracity. I think what my thought here boils down to now is that that complaint is really all about the naming of `PureComponent`. What we're doing is 100% consistent with the expectations of a React `PureComponent`, which are [documented here](https://reactjs.org/docs/react-api.html#reactpurecomponent) -- it's just that those expectations don't include all the things that you might (reasonably!) expect the word "pure" to mean, or to be true of "a pure component".

A comment about that mismatch of `PureComponent` vs. the meaning of "pure" in general would be fine to have here. (Certainly it wouldn't be the first comment in our codebase that complains about a design choice that happened somewhere else :wink: -- I've written plenty of those.) What I don't want is a warning that we're hacking around or breaking some API expectation that potentially could expose us to problems when the React implementation changes, in a spot where in fact we aren't.

Reply:
> What I don't want is a warning that we're hacking around or breaking some API expectation that potentially could expose us to problems when the React implementation changes, in a spot where in fact we aren't.

Ah, I see the ambiguity now. Amended significantly.

Reply:
New version looks great!